### PR TITLE
vscode-extensions.reditorsupport.r: init 2.8.2

### LINKED ISSUE #290471: Package request: vscode-extensions.reditorsupport.r
**Project description**

Adds R language features (e.g. syntax highlighting) to VSCode. Ideally, would also install `python311Packages.radian` and automatically configure the extension to use them. Also requires  `rPackages.languageserver` to be installed.

**Metadata**

* homepage URL: https://marketplace.visualstudio.com/items?itemName=REditorSupport.r
* source URL: https://github.com/REditorSupport/vscode-R
* license: mit
* platforms: unix, linux, darwin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by Pandapip1**
## Description of changes

Closes https://github.com/NixOS/nixpkgs/issues/290471

Adds https://github.com/REditorSupport/vscode-R

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by superherointj** [comment]
The commit title adding yourself as maintainer should be:

> maintainers: add pandapip1

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -15190,6 +15190,12 @@
     githubId = 1788628;
     name = "pandaman";
   };
+  pandapip1 = {
+    email = "gavinnjohn@gmail.com";
+    github = "Pandpaip1";
```
[the remark this answers is not in the document]
Comment (Pandapip1) [reply maintainers/maintainer-list.nix:15195]: Thanks for spotting that typo!

File: pkgs/applications/editors/vscode/extensions/reditorsupport.r/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  R,
+  rPackages,
+}:
+
+vscode-utils.buildVscodeMarketplaceExtension {
+  mktplcRef = {
+    name = "r";
+    publisher = "reditorsupport";
+    version = "2.8.2";
+    sha256 = "sha256-FPL/JjW452KRchcQ0iHXRJarZXvS3B8PvZhXjf3rMhQ=";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/reditorsupport.r/default.nix:16]:
```suggestion
    hash = "sha256-FPL/JjW452KRchcQ0iHXRJarZXvS3B8PvZhXjf3rMhQ=";
```

File: pkgs/applications/editors/vscode/extensions/reditorsupport.r/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    rPackages.languageserver
+  ];
+  postInstall = ''
+    cd "$out/$installPrefix"
+    jq '.contributes.configuration.properties."r.rpath.mac".default = "${lib.getExe' R "R"}"' package.json | sponge package.json
+    jq '.contributes.configuration.properties."r.rpath.linux".default = "${lib.getExe' R "R"}"' package.json | sponge package.json
+    jq '.contributes.configuration.properties."r.rterm.mac".default = "${lib.getExe python311Packages.radian}"' package.json | sponge package.json
+    jq '.contributes.configuration.properties."r.rterm.linux".default = "${lib.getExe python311Packages.radian}"' package.json | sponge package.json
+  '';
+  meta = {
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/reditorsupport.r/default.nix:34]:
```suggestion
  meta = {
    changelog = "https://github.com/REditorSupport/vscode-R/releases/tag/v${version}";
```

File: pkgs/applications/editors/vscode/extensions/reditorsupport.r/default.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    description = "Provides support for the R programming language";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/reditorsupport.r/default.nix:35]:
```suggestion
    description = "A Visual Studio Code extension for the R programming language";
```
Feel free to rephrase as you wish. What is important is to mention:
> A Visual Studio Code extension...

**Comment by superherointj** [comment]
@ofborg eval

**Review by superherointj — APPROVED** [review]

**Pull request merged**